src/CredentialAgent.js

```diff
@@ -0,0 +1,59 @@
+import fetch from './fetch';
+import { Login } from '.';
```

Comment:
Will this create a circular reference? `Agent -> Login -> Client -> Agent -> ...`

Reply:
Do yo mean a cycle in memory?  The imports themselves seem OK.